packages/components/src/components/checkbox-group/checkbox-group.ts

```diff
@@ -0,0 +1,176 @@
+    return [...this.querySelectorAll<SdCheckbox>('sd-checkbox')];
+  }
+
+  private async syncCheckboxElements() {
+    const checkboxes = this.getAllCheckboxes();
+
+    await Promise.all(
+      // Sync the checked state and size
+      checkboxes.map(async checkbox => {
+        await checkbox.updateComplete;
+
+        checkbox.size = this.size;
+      })
+    );
+
+    if (!checkboxes.some(checkbox => checkbox.checked)) {
+      checkboxes[0].tabIndex = 0;
+    }
+  }
+
+  private syncCheckboxes() {
+    if (customElements.get('sd-checkbox')) {
+      this.syncCheckboxElements();
+    } else {
+      customElements.whenDefined('sd-checkbox').then(() => this.syncCheckboxes());
+    }
+  }
+
+  @watch('size', { waitUntilFirstUpdate: true })
+  handleSizeChange() {
+    this.syncCheckboxes();
+  }
+  @watch('invalid', { waitUntilFirstUpdate: true })
+  handleInvalid() {
+    this.syncCheckboxes();
+  }
+
+  render() {
+    const hasLabelSlot = this.hasSlotController.test('label');
+    const hasLabel = this.label ? true : hasLabelSlot;
+    const defaultSlot = html` <slot @slotchange=${this.syncCheckboxes}></slot> `;
+
+    return html`
+      <fieldset
+        part="form-control"
+        class=${cx(
+          'form-control form-control--checkbox-group border-0 p-0 m-0',
+          hasLabel && 'form-control--has-label',
+          {
+            /* sizes, fonts */
+            sm: 'text-sm',
+            lg: 'text-base'
+          }[this.size]
+        )}
+        role="group"
+        aria-labelledby="label"
+      >
+        <label
+          part="form-control-label"
+          id="label"
+          class="form-control__label mb-2 hidden p-0 font-bold leading-normal text-black"
+          aria-hidden=${!hasLabel}
+        >
+          <slot name="label">${this.label}</slot>
+        </label>
+
+        <div
+          part="form-control-input"
+          class=${cx(
+            'form-control-input',
+            {
+              vertical: 'form-control-input--vertical flex flex-col',
+              horizontal: 'form-control-input--horizontal flex flex-row'
+            }[this.orientation]
+          )}
+        >
+          ${defaultSlot}
+        </div>
+      </fieldset>
+    `;
+  }
+
+  /**
+   * Inherits Tailwind classes and includes additional styling.
+   */
+  static styles = [
+    componentStyles,
+    SolidElement.styles,
+    css`
+      :host {
+        display: block;
+      }
+
+      .form-control-input--vertical ::slotted(sd-checkbox) {
+        margin-bottom: 8px;
+        display: flex;
+      }
+
+      .form-control-input--vertical ::slotted(sd-checkbox:last-of-type) {
+        margin-bottom: 0;
+      }
+
+      .form-control-input--horizontal ::slotted(sd-checkbox) {
+        margin-right: 24px;
+      }
+
+      :host([size='sm']) .form-control-input--horizontal ::slotted(sd-checkbox) {
+        margin-right: 16px;
+      }
+
+      .form-control-input--horizontal ::slotted(sd-checkbox:last-of-type) {
```

Comment:
Could you just check on the host if positioning is set as to omit the class `.form-control-input--horizontal`  here?